packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx

```diff
@@ -86,14 +88,21 @@ export function collectPostsFacebook(this: SocialNetworkUI) {
                 for (const url of images) {
                     info.postMetadataImages.add(url)
                     nextTypedMessage.push(makeTypedMessageImage(url))
+                    const post = await Services.Steganography.decodeImageUrl(url, {
```

Comment:
This is not right for me.

Reply:
Why?

Reply:
The collector should do its best on collecting info from the DOM nodes. Besides that, the decryption progress was handled by the  `<DecryptedPost />` component. And you should not do it twice here. The raw `decodeImageUrl` service did not cache the result for you which is already supported by the `<DecryptedPost />` component.

Reply:
Yeah, set the iv of post during decrypt image.